api/v1beta1/types.go

```diff
@@ -635,6 +640,19 @@ type ServiceEndpointSpec struct {
 	Locations []string `json:"locations"`
 }
 
+// PrivateEndpointSpec configures an Azure Private Endpoint.
+type PrivateEndpointSpec struct {
```

Comment:
This needs the flags for which fields are optional and their descriptions.

Reply:
Done.